NewCalaculator/src/MathOperations.java

```diff
@@ -0,0 +1,25 @@
+public class MathOperations {
+
+    protected int a;
+    protected int b;
+
+    public MathOperations(int a, int b) {
+        this.a = a;
+        this.b = b;
+    }
+
+    public MathOperations() {}
+
+    public double calculate(MathOperations newMathematicsOperation) {
+        return newMathematicsOperation.calculateParameters(a, b);
```

Comment:
Bardzo dobry kierunek z tą klasą, ale jednak troszkę wydaje się to przekombinowane.

Zamiast klasy, którą wszystkie te operacje dziedziczą to lepszą opcją byłoby jednak zrobienie interfejsu "MathOperations".

Patrz co aktualnie sie tutaj dzieje - MathOperations ma dwa pola:
![image](https://user-images.githubusercontent.com/17989854/151667234-53faabeb-74c5-48b3-b68f-952f680faaa0.png)


Przykładowa operacja **dziedziczy** MathOperations, a wraz z nim te dwa pola, a mimo to przyjmuje jako argument dwa pola:

![image](https://user-images.githubusercontent.com/17989854/151667256-1e325ea0-611f-41c4-ac36-63a0141e7c48.png)



Wydaje się, że albo 

a) argumenty w metodzie calculateParameters są zbędne (bo przecież możesz sobie w klasię Addition zawołać this.a / this.b i dostać się do pól z klasy MathOperations)

b) pola w klasie MathOperatoions są zbędne.

Osobiście dla mnie bardziej opcja b brzmi sensowniej co powoduje, że może warto byłoby je usunąć.

Wtedy zostaje nam tak na prawdę klasa, która ma metodę 'calculateParameters' i przyjmuje argument a i b. A skoro to tylko jedna metoda (a w dodatku nie chcesz jej implementować w klasie MathOperations) to można równie dobrze MathOperations zrobić interfejsem z tą jedną metodą.

Reply:
Nie chciałbym rezygnować całkowicie z  klasy MathOperations bo wtedy rozwali mi to pozostałą część z wyborem operacji  w metodzie main :(

![image](https://user-images.githubusercontent.com/95712997/151719483-d5587ead-8863-4667-919e-7702c8b50e4f.png)